app/api/v1/farmers.py

```
@@ -0,0 +1,88 @@
+    """Get farmer by user ID."""
+    farmer_service = FarmerService(db)
+    farmer = await farmer_service.get_farmer_by_user_id(user_id)
+    if not farmer:
+        raise HTTPException(status_code=404, detail="Farmer profile not found for this user")
+    return farmer
+
+@router.put("/{farmer_id}", response_model=FarmerResponse)
+async def update_farmer(
+    farmer_id: int,
+    farmer_data: FarmerUpdate,
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.11.9)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)